src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -165,11 +165,22 @@ protected override void OnKeyUp(KeyEventArgs e)
                             {
                                 // Search for the best version
                                 NuGetVersion rangeBestVersion = userRequestedVersionRange.FindBestMatch(versions);
-                                bool isBestOption = rangeBestVersion.ToString() == _versions.Items[_versions.SelectedIndex].ToString();
+                                DisplayVersion selectedVersion = (DisplayVersion)_versions.Items.CurrentItem;
+                                bool isBestOption = rangeBestVersion.ToString() == selectedVersion.Version.ToString();
                                 if (isBestOption)
                                 {
-                                    PackageDetailControlModel.SelectedVersion = new DisplayVersion(userRequestedVersionRange, rangeBestVersion, additionalInfo: null, isVulnerable: false);
-                                    _versions.Text = comboboxText;
+                                    // Add vulnerable/deprecated labels to non floating/range versions
+                                    if (userRequestedVersionRange.OriginalString.StartsWith("(", StringComparison.OrdinalIgnoreCase) ||
+                                        userRequestedVersionRange.OriginalString.StartsWith("[", StringComparison.OrdinalIgnoreCase) ||
+                                        userRequestedVersionRange.IsFloating)
```

Comment:
The `IsFloating` check is probably a tiny bit cheaper than the string check, so move it up?

```suggestion
                                    if (userRequestedVersionRange.IsFloating ||
                                        userRequestedVersionRange.OriginalString.StartsWith("(", StringComparison.OrdinalIgnoreCase) ||
                                        userRequestedVersionRange.OriginalString.StartsWith("[", StringComparison.OrdinalIgnoreCase))
```